File: 2022/221227-dojo/java/app/src/main/java/Solution.java
class Solution {
    public void solution(int N) {
        int enable_print = N % 10;
        while (N > 0) {
            if (enable_print == 0 && N % 10 != 0) {
                enable_print = 1;
                N = N / 10;
                continue;
            }
            else if (enable_print >= 1) {
                System.out.print(N % 10);
            }
            N = N / 10;
        }
    }
}

//
//
//class Solution {
//    public void solution(int N) {
//        int enable_print = N % 10;
//        while (N > 0) {
//            if (enable_print == 0 && N % 10 != 0) {
//                enable_print = 1;
//            }
//            else if (enable_print == 1) {
//                System.out.print(N % 10);
//            }
//            N = N / 10;
//        }
//    }
//}

File: 2022/221227-dojo/javascript/solution.js
export function makeArray(A) {
  return A.sort((a, b) => b - a)
    .map((value) => value.toString().charAt(0) + (value % 10).toString());
}

export function validatorToFindDuplicated(array) {
  const validator = new Set();

  array.forEach((value) => validator.add(value));
  if (validator.size === array.length) {
    return -1;
  }

  return 1;
}

export default function solution(A) {
  const array = makeArray(A);

  const result = validatorToFindDuplicated(array);

  if (result === -1) {
    return result;
  }

  const resultArray = [];

  for (let i = 0; i < array.length; i += 1) {
    for (let j = i + 1; j < array.length; j += 1) {
      if (array[i] === array[j]) {
        resultArray.push(A.sort((a, b) => b - a)[i] + A.sort((a, b) => b - a)[j]);
      }
    }
  }

  return resultArray.sort((a, b) => b - a)[0];
}
